providers/song-controls-front.js

```diff
@@ -1,18 +1,43 @@
 const { ipcRenderer } = require("electron");
 
-let videoStream = document.querySelector(".video-stream");
+let video = document.querySelector("video");
+
 module.exports = () => {
-    ipcRenderer.on("playPause", () => {
-        if (!videoStream) {
-            videoStream = document.querySelector(".video-stream");
-        }
+    ipcRenderer.on("playPause", (_e, toPlay) => playPause(toPlay));
+};
+module.exports.playPause = playPause;
 
-        if (videoStream.paused) {
-            videoStream.play();
-        } else {
-            videoStream.yns_pause ?
-                videoStream.yns_pause() :
-                videoStream.pause();
+function playPause(toPlay = undefined) {
+    if (!checkVideo()) return;
+    
+    switch(toPlay) {
+        case undefined: 
+            video.paused ? play() : pause();
+            break;
+        case true:
+            play(); 
+            break;
+        case false:
+            pause();
+    }
+
+    function play() {
+        video.play();
+    }
+    
+    function pause() {
+        video.yns_pause ?
+            video.yns_pause() :
+            video.pause();
+    }
+    
+}
+
+function checkVideo() {
+    if (!video) {
+        video = document.querySelector("video");
+        if (!video) {
+            return false;
         }
-    });
-};
+    } return true;
```

Comment:
nit: it can be simplified into
```js
function checkVideo() {
    if (!video) {
        video = document.querySelector("video");
    }

    return !!video;
}
```

Reply:
done 👍🏼